File: app/schemas/food_item.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import date, datetime
import uuid


class FoodItemBase(BaseModel):
    name: str
    barcode: Optional[str] = None
    category: Optional[str] = None
    quantity: int = 1
    expiration_date: Optional[date] = None
    image_url: Optional[str] = None
    source: str = "manual"  # barcode|vision|manual


class FoodItemCreate(FoodItemBase):
    pass


class FoodItemUpdate(BaseModel):
    name: Optional[str] = None
    barcode: Optional[str] = None
    category: Optional[str] = None
    quantity: Optional[int] = None
    expiration_date: Optional[date] = None
    image_url: Optional[str] = None


class FoodItem(FoodItemBase):
    id: uuid.UUID
    added_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/foodItem.ts
export interface FoodItem {
  id: string;
  name: string;
  category?: string;
  quantity?: number;
  unit?: string;
  expiration_date?: string;
  barcode?: string;
  image_url?: string;
  notes?: string;
  source?: string;
  created_at: string;
  updated_at: string;
  owner_id: string;
}

export interface FoodItemCreate {
  name: string;
  category?: string;
  quantity?: number;
  unit?: string;
  expiration_date?: string;
  barcode?: string;
  image_url?: string;
  notes?: string;
  source?: string;
}

export interface FoodItemUpdate {
  name?: string;
  category?: string;
  quantity?: number;
  unit?: string;
  expiration_date?: string;
  barcode?: string;
  image_url?: string;
  notes?: string;
}
